core/server/web/shared/middlewares/custom-redirects.js

```diff
@@ -54,17 +54,19 @@ _private.registerRoutes = () => {
             customRedirectsRouter.get(new RegExp(redirect.from, options), function (req, res) {
                 const maxAge = redirect.permanent ? config.get('caching:customRedirects:maxAge') : 0;
                 const toURL = url.parse(redirect.to);
-                const currentURL = url.parse(req.originalUrl);
+                const currentURL = url.parse(req.url);
 
-                // if the URL points to an external website, remove Ghost's base path
-                /** @see https://github.com/TryGhost/Ghost/issues/10776 */
-                const currentPath = (toURL.hostname)
-                    ? currentURL.pathname.replace(urlUtils.getSubdir(), '')
-                    : currentURL.pathname;
-
-                toURL.pathname = currentPath.replace(new RegExp(redirect.from, options), toURL.pathname);
+                toURL.pathname = currentURL.pathname.replace(new RegExp(redirect.from, options), toURL.pathname);
                 toURL.search = currentURL.search;
 
+                /**
+                 * Only if the URL is internal should we prepend the Ghost subdirectory
+                 * @see https://github.com/TryGhost/Ghost/issues/10776
+                 */
+                if (!toURL.hostname) {
+                    toURL.pathname = path.join(urlUtils.getSubdir(), toURL.pathname);
```

Comment:
It seems incorrect to use the `path` module here - it's for directories, not URLs. :shrug: Not sure if I should refactor this out, would make the code more complicated if so (having to check we don't duplicate slashes, or worse, leave one out :scream:)

Reply:
It's totally incorrect, but isn't this just a case of doing `urlUtils.urlJoin` instead?

Reply:
Ah - forgot about that :+1: